lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -459,11 +464,93 @@ bool SessionState::apply_update_criteria(SessionStateUpdateCriteria& uc) {
   return true;
 }
 
+optional<RuleStats> SessionState::get_rule_delta(
+    const std::string& rule_id, uint64_t rule_version, uint64_t used_tx,
+    uint64_t used_rx, uint64_t dropped_tx, uint64_t dropped_rx,
+    SessionStateUpdateCriteria& uc) {
```

Comment:
Can you change the type of `SessionStateUpdateCriteria& uc` to `SessionStateUpdateCriteria* session_uc`? 
(Trying to conform a bit more to the google c++ guide, which specifies output parameters should be pointers)